### PR TITLE
Pt 163588147 merge master to minerva

### PR DESCRIPTION
**Pull request by lucafavatella**

### CODE REVIEW

**Comment by lucafavatella** [comment]
System test sync [failed on CI](https://circleci.com/gh/aeternity/aeternity/7504):
```
%%% aest_sync_SUITE ==> new_node_joins_network: FAILED
%%% aest_sync_SUITE ==> {timeout,[{aest_nodes,wait_for_value,6,
                      [{file,"/home/circleci/aeternity/_build/system_test+test/extras/system_test/helpers/aest_nodes.erl"},
                       {line,806}]},
          {aest_nodes,loop_for_values,6,
                      [{file,"/home/circleci/aeternity/_build/system_test+test/extras/system_test/helpers/aest_nodes.erl"},
                       {line,768}]},
          {aest_sync_SUITE,new_node_joins_network,1,
                           [{file,"/home/circleci/aeternity/_build/system_test+test/extras/system_test/aest_sync_SUITE.erl"},
                            {line,237}]},
```
that means that new_node1 did not sync fast enough.

Excerpts from [test log](https://7504-99802036-gh.circle-artifacts.com/0/home/circleci/aeternity/system_test/logs/ct_run.aeternity_ct@localhost.2019-02-14_17.08.32/extras.system_test.aest_sync_SUITE.logs/run.2019-02-14_17.08.33/aest_sync_suite.new_node_joins_network.html) that shows that waiting time was 4s:
```
*** User 2019-02-14 17:08:46.195 ***
Reached after 5.59 seconds (slack 84410 ms) Height 31 on nodes [old_node1,
                                                                old_node2]

*** User 2019-02-14 17:08:46.195 ***
Mining time per block 266 ms for 30 blocks

...

*** User 2019-02-14 17:08:51.379 ***
Reached after 3.59 seconds (slack 16410 ms) Height 0 on nodes [new_node1]

*** User 2019-02-14 17:08:51.380 ***
GET "http://localhost:35049/v2/key-blocks/height/30"

...

*** CT Error Notification 2019-02-14 17:08:55.435 ***
aest_nodes:wait_for_value failed on line 806
Reason: timeout
```

In [sync log of node](https://7504-99802036-gh.circle-artifacts.com/0/home/circleci/aeternity/system_test/logs/ct_run.aeternity_ct%40localhost.2019-02-14_17.08.32/extras.system_test.aest_sync_SUITE.logs/run.2019-02-14_17.08.33/log_private.1/new_node1_logs/aeternity_sync.log):
```
2019-02-14 17:08:50.680 [debug] <0.1480.0>@aec_sync:do_work_on_sync_task:656 Agreed upon height ("pp_qfqV5C8...adryAwq"): 0
2019-02-14 17:08:51.526 [info] <0.1480.0>@aec_sync:post_blocks:685 Synced blocks 1 - 2
2019-02-14 17:08:58.923 [info] <0.1480.0>@aec_sync:post_blocks:685 Synced blocks 3 - 38
```

It seems [known](https://www.pivotaltracker.com/story/show/163093465).

**Comment by lucafavatella** [comment]
I proposed additional log for better debugging next time.

**Comment by lucafavatella** [comment]
I checked the [mining log](https://7504-99802036-gh.circle-artifacts.com/0/home/circleci/aeternity/system_test/logs/ct_run.aeternity_ct%40localhost.2019-02-14_17.08.32/extras.system_test.aest_sync_SUITE.logs/run.2019-02-14_17.08.33/log_private.1/new_node1_logs/aeternity_mining.log) - it seems the posting of blocks to the conductor is quite constant over time so it may not be worker sleeping but simply time needed for adding block:
```
...
...[info] <0.1441.0>@aec_conductor:handle_synced_block:953 synced_block:...
...
```

**Comment by lucafavatella** [comment]
I see [CI job smoke system tests failed](https://circleci.com/gh/aeternity/aeternity/7537):
```
%%% aest_sync_SUITE ==> stop_and_continue_sync: FAILED
%%% aest_sync_SUITE ==> {{node1,{error,socket_closed_remotely}},
 [{aest_nodes,get_block,2,
              [{file,"/home/circleci/aeternity/_build/system_test+test/extras/system_test/helpers/aest_nodes.erl"},
               {line,372}]},
  {aest_sync_SUITE,stop_and_continue_sync,1,
                   [{file,"/home/circleci/aeternity/_build/system_test+test/extras/system_test/aest_sync_SUITE.erl"},
                    {line,350}]},
```

File: apps/aecore/src/aec_sync.erl [hunk 1]
```diff
@@ -330,6 +330,7 @@ get_next_work_item(ST = #sync_task{ pool = [{_, _, {_, _}} | _] = Pool, adding =
         _ when length(Pend) < 10 orelse NewPool /= [] ->
             get_next_work_item(ST#sync_task{ pool = NewPool, pending = Pend ++ [ToBeAdded] });
         _ ->
+            epoch_sync:info("Nothing to do", []),
```
Comment (hanssv) [inline apps/aecore/src/aec_sync.erl]: Wait, where did this come from....

Comment (hanssv) [reply apps/aecore/src/aec_sync.erl]: No, I don't like this one. But I can perhaps be convinced.

Comment (lucafavatella) [reply apps/aecore/src/aec_sync.erl]: Removed.

Comment (lucafavatella) [reply apps/aecore/src/aec_sync.erl]: Context was https://github.com/aeternity/aeternity/pull/2159#issuecomment-463761595

**Pull request merged**